numpy/_core/strings.py

```diff
@@ -139,15 +139,40 @@ def multiply(a, i):
            ['d', 'ee', 'fff']], dtype='<U3')
            
     """
-    a_arr = np.asarray(a)
-    i_arr = np.asarray(i)
-    if not issubclass(i_arr.dtype.type, np.integer):
-        raise ValueError("Can only multiply by integers")
-    if a_arr.dtype.kind == "T":
-        return a_arr * i_arr
-    out_size = _get_num_chars(a_arr) * max(int(i_arr.max()), 0)
-    return _vec_string(
-        a_arr, type(a_arr.dtype)(out_size), '__mul__', (i_arr,))
+    a = np.asanyarray(a)
```

Comment:
The Python code for this has gotten a bit big, and I'm wondering whether I should reconsider the overflow check etc. Maybe we should just let the buffer allocation handle that. I'm obviously open to suggestions.

Reply:
Can't you do the overflow check in C? Something like https://github.com/numpy/numpy/blob/main/numpy/_core/src/umath/stringdtype_ufuncs.cpp#L133-L142.

Reply:
The thing is we need to preallocate the buffer in Python, like we do for others as well, cause we can't figure out the buffer size otherwise.

Reply:
Ah in that case there's not much we can do I think? 

One option would be to just ignore this problem and force people to specify output dtypes that are large enough to hold the result, lest they truncate, but that's not terribly user-friendly. That is more in the spirit of making the functions in `np.strings` lightweight wrappers around the ufuncs. You'd probably need to move all this buffer size checking into the `np.char` implementation though for backward compatibility.

Also see the other comment about stringdtype, which should allow you to delete some of this.

Reply:
I feel like going with the user friendly option is the way to do it. Now that I think about it, other ufuncs like `replace` has similar wrappers around them. In essense it's always about setting up parameters, do some checks, compute output buffer size, call ufunc, which is the case here as well. So if you agree, let's leave it as is.